src/fauxtoshop.cpp

```diff
@@ -1,517 +1,492 @@
+ * Params: img (GBufferedImage)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: nothing. Void function
+*/
+void applyScatterFilter(GBufferedImage &img) {
+    // Convert your GBufferedImage object into a Grid<int>
+    Grid<int> original = img.toGrid();
+    Grid<int> scatterGrid(original.numRows(), original.numCols());
+    // Prompt user for scatter radius input between 1 and 100 (inclusive)
+    int scatterRadius = getValidInteger(
+                "Enter the scatter radius as number between 1 and 100: ", 1, 100);
+    // Iterate through each pixel in the image grid
+    for (int i = 0; i < scatterGrid.numRows(); i++) {
+        for (int j = 0; j < scatterGrid.numCols(); j++) {
+            // Randamly select nearby pixel (row and col) to provide new colour
+            getRandomPixelColor(original, scatterGrid, i, j, scatterRadius);
+        }
+    }
+    img.fromGrid(scatterGrid);
+}
+
+/*
+ * Function: applyEdgeDetectionFilter
+ * Usage: Create a new black and white image the same size as the original. A pixel is black if it
+ * is an edge or white if it is not an edge
+ * Function:
+ * a. Prompt user to provide an edge threshold that specifies the difference between 2 pixels
+ * (positive integer) before they make an 'edge'. If input is invalid reprompt
+ * b. Loop over each pixel to determine whether it is an edge or not
+ * c. Check all 9 pixels (including self at 0) to check if at least 1 neighbour has a difference
+ * greater than the threshold e.g. the pixel is an edge for [9, 8, 5, 3, 3, 0 (self), 4, 7, 8, 7]
+ * if threshold < 9
+ * Params: img (GBufferedImage)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: nothing. Void function
+*/
+void applyEdgeDetectionFilter(GBufferedImage &img) {
+    // Prompt user for scatter radius input as a positive integer
+    int edgeThreshold = getValidInteger(
+                "Enter the edge threshold as a positive number: ", 0);
+    // Set up a new image grid for the black and white output for edge detection
+    Grid<int> original = img.toGrid();
+    Grid<int> blackWhiteGrid(original.numRows(), original.numCols());
+    // Iterate through each pixel
+    for (int i = 0; i < original.numRows(); i++) {
+        for (int j = 0; j < original.numCols(); j++) {
+            // Change the pixel to black or white depending on whether it is an edge
+            blackWhiteGrid[i][j] = isEdge(original, edgeThreshold, i, j);
+        }
+    }
+    img.fromGrid(blackWhiteGrid);
+}
+
+/*
+ * Function: isEdge
+ * Usage: Calculates whether a pixel is an edge
+ * Params: original (Grid of integers, edgeThreshold (integer), row (integer), col (integer)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: black or white colour (integer)
+*/
+int isEdge(const Grid<int> &original, int edgeThreshold, int row, int col) {
```

Comment:
Nice! Tiny thing to make this more readable: the name `isEdge` sounds like this function should returns a `bool`, but it's actually returning the pixel value. Maybe you can either make this true/false and then in the parent method you can do `blackWhiteGrid[i][j] = isEdge(original, edgeThreshold, i, j) ? BLACK : WHITE;`, or even simpler you can just rename this method to something more descriptive.